File: jsonParsing/app/src/main/java/com/example/jsonparsing/ResponseAdapter.java
package com.example.jsonparsing;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class ResponseAdapter extends RecyclerView.Adapter<ResponseAdapter.ResponseViewholder> {

    ArrayList<Response> responses ;

    public ResponseAdapter(ArrayList<Response> responses) {
        this.responses = responses;
    }

    @NonNull
    @Override
    public ResponseViewholder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater li = (LayoutInflater) parent.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View itemView = li.inflate(R.layout.string_row,parent,false);
        return new ResponseViewholder(itemView);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull ResponseViewholder holder, int position) {
        holder.tvName.setText(responses.get(position).getUser().getName().getFirst() + responses.get(position).getUser().getName().getLast());
        holder.tvString.setText(responses.get(position).getText());
    }

    @Override
    public int getItemCount() {
        return responses.size();
    }


    class ResponseViewholder extends RecyclerView.ViewHolder{
        TextView tvString,tvName;

        public ResponseViewholder(@NonNull View itemView) {
            super(itemView);
            tvName = itemView.findViewById(R.id.tvName);
            tvString = itemView.findViewById(R.id.tvString);
        }
    }
}

File: jsonParsing/app/src/main/java/com/example/jsonparsing/ResponseAdapter.kt
//package com.example.jsonparsing
//
//import android.content.Context
//import android.util.Log
//import android.view.LayoutInflater
//import android.view.View
//import android.view.ViewGroup
//import androidx.recyclerview.widget.RecyclerView
//import kotlinx.android.synthetic.main.string_row.view.*
//
//open class ResponseAdapter(responses: ArrayList<Response>) : RecyclerView.Adapter<ResponseAdapter.JsonViewHolder>() {
//
//    val responses  = arrayListOf<Response>()
//
//    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): JsonViewHolder {
//        val layoutInflater : LayoutInflater = parent.context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
//        val view = layoutInflater.inflate(R.layout.string_row,parent,false)
//        return JsonViewHolder(view)
//    }
//
//    override fun onBindViewHolder(holder: JsonViewHolder, position: Int) {
//        holder.tvName.text = responses[position].user.name.first
//        holder.tvString.text = responses[position].text
//    }
//
//    override fun getItemCount(): Int {
//        return responses.size
//    }
//
//    class JsonViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
//        val tvString = itemView.tvString
//        val tvName = itemView.tvName
//    }
//
//}